### ISSUE TITLE
Cloudflare: Environment variables starting with `NITRO_` are not passed to runtimeConfig

### ISSUE THREAD

**Issue by MickL** [nitrojs/nitro#2468]
### Environment

nitropack: 2.9.6

### Reproduction

-

### Describe the bug

Environment variables dont seem to work on Cloudflare:

```ts
export default defineNitroConfig({
  runtimeConfig: {
    myVar: ''
  },
});
```

This works locally with .env:
```
NITRO_MY_VAR="Hello world"
```

But deployed to Cloudflare with the environment set in Cloudflare Dashboard it stays empty. Using the following workaround it works:

```ts
  runtimeConfig: {
    myVar: process.env.NITRO_MY_VAR,
  },
```

But what would then the purpose of defining `NITRO_` env vars in the first place when they are not auto "injected"?

**Comment by pi0** [comment]
Do you call `runtimeConfig(event)` inside an event handler? Due to how late cloudflare provides environment (during request) it is a limitation.

**Comment by MickL** [comment]
I do not. It works with the example above so in nuxt config the process env is there?

**Comment by te1** [comment]
See also https://github.com/Atinux/nuxt-env-repro which gives this output (called from inside an event handler):

```json
{
  "runtimeConfig().hello": "",
  "runtimeConfig(event).hello": "world",
  "process.env.NUXT_HELLO": "world"
}
```

**Comment by jamaluddinrumi** [comment]
might be you want to ensure `NITRO_` env vars are passed to runtimeConfig by reassign them on `request` hook.

i'm facing only my env vars are passed to runtimeConfig. 

and i reassign them to ensure they are all passed and voila, they are there all inside runtimeConfig.

reference https://github.com/nitrojs/nitro/pull/2575#issuecomment-2480455926

**Comment by chrisspiegl** [comment]
When defining the runtimeConfig with `process.env.NITRO_` variables… they don't get run on execution but rather are put right into the build process (and then deployed)/
```
runtimeConfig: {
  myVar: process.env.NITRO_MY_VAR,
},
```

This leads to the variable value being part of the deploy rather than being read from Clouflare… that's why it appears to be "working" but in reality, if you run a differen `.env` variable in Cloudflare it would not be updated.

I checked this by searching for the variable value in the `dist` folder.

**Comment by MickL** [comment]
I dont think you should use any .env file on Cloudflare or other platforms like Render or Vercel? Instead you define the variables in the env settings of the worker.

**Comment by chrisspiegl** [comment]
This is only true if you deploy via Github => Cloudflare Pages.

If you run a local `pnpm build` followed by `pnpm wrangler pages deploy` you will end up building the variable into the code.

And my statement from before still holds true:

If you do it this way, the value of your environment variable is built into the code and if you change it on Cloudflare's dashboard, it will not end up in the current build until you do a re-deployment.
